e2e/setup-server.ts

```diff
@@ -0,0 +1,36 @@
+import execa, { ExecaChildProcess } from 'execa'
+import { getRandomPort, waitForPort } from 'get-port-please'
+
+export type ServerContext = {
+  serverProcess: ExecaChildProcess
+  url: (val: string) => string
+}
+export async function startServer(): Promise<ServerContext> {
+  const host = '127.0.0.1'
+  const port = await getRandomPort(host)
+
+  // console.log('building...')
+  // await execa('pnpm', ['--filter=example', 'build'], {
+  //   stdio: 'inherit',
+  //   env: {
+  //     ...process.env,
+  //     _PORT: String(port),
+  //     PORT: String(port),
+  //     HOST: host
+  //   }
+  // })
+  // console.log('build finished!')
+
+  const serverProcess = execa(
```

Comment:
I think we should use `spawn` of `child_process`. 
Because e2e should run in no special environment like windows. If so, you should use `execa` which abstracts the environment.